src/common/btree.c

```diff
@@ -1102,7 +1113,11 @@ btr_node_split_and_insert(struct btr_context *tcx, struct btr_trace *trace,
 		/* insert the right node and the first key of the right
 		 * node to its parent
 		 */
-		if (btr_is_direct_key(tcx))
+		if (btr_is_direct_key(tcx) && btr_is_csum_enabled(tcx)) {
+			rec->rec_hnode->rec_node = off_right;
+			btr_hkey_copy(tcx, &rec->rec_hnode->rec_hkey[0],
+				      &rec_dst->rec_hnode->rec_hkey[0]);
+		} else if (btr_is_direct_key(tcx))
```

Comment:
can check btr_is_direct_key once if you move the csum specific stuff into the if block

Reply:
indeed